qmi/instruments/thorlabs/mpc320.py

```diff
@@ -0,0 +1,507 @@
+    jog_step1: float
+    jog_step2: float
+    jog_step3: float
+
+
+Thorlabs_MPC320_ChannelMap: Dict[int, int] = {1: 0x01, 2: 0x02, 3: 0x04}
+
+
+class Thorlabs_MPC320(QMI_Instrument):
+    """
+    Driver for a Thorlabs MPC320 motorised fibre polarisation controller.
+    """
+
+    DEFAULT_RESPONSE_TIMEOUT = 1.0
+
+    # the maximum range for a paddle is 170 degrees
+    # the value returned by the encoder is 1370 for 170 degrees
+    ENCODER_CONVERSION_UNIT = 170 / 1370
+
+    MIN_POSITION_DEGREES = 0
+    MAX_POSITION_DEGREES = 170
+
+    MIN_VELOCITY_PERC = 10
+    MAX_VELOCITY_PERC = 100
+
+    MIN_CHANNEL_NUMBER = 1
+    MAX_CHANNEL_NUMBER = 3
+
+    def __init__(self, context: QMI_Context, name: str, transport: str) -> None:
+        """Initialize the instrument driver.
+
+        Parameters:
+            name:       Name for this instrument instance.
+            transport:  QMI transport descriptor to connect to the instrument.
+        """
+        super().__init__(context, name)
+        self._transport = create_transport(transport, default_attributes={"baudrate": 115200, "rtscts": True})
+        self._apt_protocol = AptProtocol(self._transport, default_timeout=self.DEFAULT_RESPONSE_TIMEOUT)
+
+    def _validate_position(self, pos: float) -> None:
+        """
+        Validate the position. Any position for the MPC320 needs to be in the range 0 to 170 degrees,
+        or 0 to 1370 in encoder counts.
+
+        Parameters:
+            pos:    Position to validate in degrees.
+
+        Raises:
+            an instance of QMI_InstrumentException if the position is invalid.
+        """
+        if not self.MIN_POSITION_DEGREES <= pos <= self.MAX_POSITION_DEGREES:
+            raise QMI_InstrumentException(
+                f"Given position {pos} is outside the valid range \
+                    [{self.MIN_POSITION_DEGREES}, {self.MAX_POSITION_DEGREES}]"
+            )
+
+    def _validate_velocity(self, vel: float) -> None:
+        """
+        Validate the velocity. Any velocity for the MPC320 needs to be in the range 40 to 400 degrees/s,
+        or 10 to 100% of 400 degrees/s.
+
+        Parameters:
+            vel:    Velocity to validate in percentage.
+
+        Raises:
+            an instance of QMI_InstrumentException if the velocity is invalid.
+        """
+        if not self.MIN_VELOCITY_PERC <= vel <= self.MAX_VELOCITY_PERC:
+            raise QMI_InstrumentException(
+                f"Given relative velocity {vel} is outside the valid range \
+                    [{self.MIN_VELOCITY_PERC}%, {self.MAX_VELOCITY_PERC}%]"
+            )
+
+    def _validate_channel(self, channel_number: int) -> None:
+        """
+        Validate the channel number. The MPC320 has 3 channels.
+
+        Parameters:
+            channel_number: Channel number to validate.
+
+        Raises:
+            an instance of QMI_InstrumentException if the channel is not 1, 2 or 3
+        """
+
+        if channel_number not in [1, 2, 3]:
```

Comment:
Minor: If you want this to be more flexible, you could do `range(self.MIN_CHANNEL_NUMBER, self.MAX_CHANNEL_NUMBER + 1)`. So it can be directly applied also for 2 channel cases, if that gets implemented.